platform.txt

```diff
@@ -172,3 +172,11 @@ tools.serial_upload.path.linux64={runtime.hardware.path}/tools/linux64
 tools.serial_upload.upload.params.verbose=-d
 tools.serial_upload.upload.params.quiet=n
 tools.serial_upload.upload.pattern="{path}/{cmd}" {serial.port.file} {upload.altID} {upload.usbID} "{build.path}/{build.project_name}.bin"
+
+# blackmagic upload for generic STM32
+tools.bmp_upload.cmd=arm-none-eabi-gdb
+tools.bmp_upload.path={runtime.tools.arm-none-eabi-gcc-6-2017-q2-update.path}/bin/
```

Comment:
Based on @facchinm comment  in https://github.com/arduino/Arduino/issues/7616#issuecomment-390894992
`{runtime.tools.arm-none-eabi-gcc}` will be used by default in git repo.
I will update path when releasing the package to set the gcc version as it is done for other upload tools.